numpy/core/tests/test_multiarray.py

```diff
@@ -4852,38 +4861,32 @@ def test_fromstring_count0(self):
         d = np.fromstring("1,2", sep=",", dtype=np.int64, count=0)
         assert d.shape == (0,)
 
-    def test_empty_files_binary(self):
+    def test_empty_files_text(self, tmp_filename):
         with open(self.filename, 'w') as f:
             pass
         y = np.fromfile(self.filename)
         assert_(y.size == 0, "Array not empty")
```

Comment:
I would suggest a deeper refactoring, using pytest's `tmp_file` with no extra fixtures needed in the class setup or teardown. Doesn't this work:

```suggestion
    def test_empty_files_text(self, tmp_file):
        with open(tmp_file, 'w') as f:
            pass
        y = np.fromfile(tmp_file)
        assert_(y.size == 0, "Array not empty")
```

Reply:
@mattip, unless I'm mistaken, `tmp_file` is not a built-in pytest fixture. You can try by running this dummy test:

```python
def test_foobar(tmp_file):
    pass
```

It will fail because the fixture does not exist.

Besides, if such built-in fixture existed, it would probably also create the file on disk which is something we don't want for the tests. We only want a filename without pre-creating the file.

Reply:
Ahh, my bad. It is `tmp_path` not `tmp_file` which gives you a PathLib object pointing to a directory like `/tmp/pytest-of-matti/pytest-8/test_foobar0` where each invocation of pytest will increment the `pytest-8` part and delete old ones according to [some logic](https://docs.pytest.org/en/latest/how-to/tmpdir.html#the-default-base-temporary-directory), and `test_foobar0` is the name of the test with a 0 for good measure. So this test would be
```
def test_empty_files_text(self, tmp_path):
        with open(tmp_path / 'empty', 'w') as f:
            pass
        y = np.fromfile(tmp_path / 'empty')
        assert_(y.size == 0, "Array not empty")
```

Reply:
Ok, but if `tmp_path` is used directly, then all references to `self.filename` must be removed. Another benefit of the custom fixture `tmp_filename` is that it is parametrized to cover two cases with very low effort (`filename` is a `str` and `filename` is a path-like object). Doing so allows to combine at least two tests into one test.